packages/frontend/core/src/components/affine/setting-modal/general-setting/about/update-check-section.tsx

```diff
@@ -0,0 +1,168 @@
+import { SettingRow } from '@affine/component/setting-components';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { Button } from '@toeverything/components/button';
+import {
+  downloadProgressAtom,
+  isCheckingForUpdatesAtom,
+  updateAvailableAtom,
+  updateReadyAtom,
+  useAppUpdater,
+} from '@toeverything/hooks/use-app-updater';
+import clsx from 'clsx';
+import { useAtomValue } from 'jotai';
+import { useCallback, useMemo, useState } from 'react';
+
+import { Loading } from '../../../../pure/workspace-slider-bar/workspace-card/loading-icon';
+import * as styles from './style.css';
+
+enum CheckUpdateStatus {
+  UNCHECK = 'uncheck',
+  LATEST = 'latest',
+  UPDATE_AVAILABLE = 'update-available',
+  ERROR = 'error',
+}
+
+const useUpdateStatusLabels = (checkUpdateStatus: CheckUpdateStatus) => {
+  const t = useAFFiNEI18N();
+  const isCheckingForUpdates = useAtomValue(isCheckingForUpdatesAtom);
+  const updateAvailable = useAtomValue(updateAvailableAtom);
+  const updateReady = useAtomValue(updateReadyAtom);
+  const downloadProgress = useAtomValue(downloadProgressAtom);
+
+  const buttonLabel = useMemo(() => {
+    return (
+      [
+        !!updateAvailable &&
+          downloadProgress === null &&
+          t['com.affine.aboutAFFiNE.checkUpdate.button.download'](),
+        !!updateReady &&
+          t['com.affine.aboutAFFiNE.checkUpdate.button.restart'](),
+        (checkUpdateStatus === CheckUpdateStatus.LATEST ||
+          checkUpdateStatus === CheckUpdateStatus.ERROR) &&
+          t['com.affine.aboutAFFiNE.checkUpdate.button.retry'](),
+      ].find(Boolean) || t['com.affine.aboutAFFiNE.checkUpdate.button.check']()
+    );
+  }, [checkUpdateStatus, downloadProgress, t, updateAvailable, updateReady]);
+
+  const subtitleLabel = useMemo(() => {
+    return (
+      [
+        !!updateAvailable &&
+          downloadProgress === null &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.update-available']({
+            version: updateAvailable.version,
+          }),
+        isCheckingForUpdates &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.checking'](),
+        !!updateAvailable &&
+          downloadProgress !== null &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.downloading'](),
+        !!updateReady &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.restart'](),
+        checkUpdateStatus === CheckUpdateStatus.ERROR &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.error'](),
+        checkUpdateStatus === CheckUpdateStatus.LATEST &&
+          t['com.affine.aboutAFFiNE.checkUpdate.subtitle.latest'](),
+      ].find(Boolean) ||
+      t['com.affine.aboutAFFiNE.checkUpdate.subtitle.check']()
+    );
+  }, [
+    checkUpdateStatus,
+    downloadProgress,
+    isCheckingForUpdates,
+    t,
+    updateAvailable,
+    updateReady,
+  ]);
+
```

Comment:
Please just use plain if/else. It does not increase code readability at all writing this way.